src/test/java/com/google/sps/MemeDatabaseTest.java

```diff
@@ -0,0 +1,89 @@
+package com.google.sps;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.google.sps.data.Meme;
+import com.google.sps.data.MemeDatabase;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Responsible for testing that the MemeDatabase correctly stores and retreives entities */
+@RunWith(JUnit4.class)
+public final class MemeDatabaseTest {
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private MemeDatabase memeDatabase;
+  private static final Gson gson = new Gson();
+  private static final String ID_1 = "123456";
+  private static final String ID_2 = "234567";
+  private static final String ID_3 = "345678";
+  private static final String SOURCE_1 = "images/dir/source1.png";
+  private static final String SOURCE_2 = "images/source2.png";
+  private static final String SOURCE_3 = "images/source3.png";
+  private static final String TITLE_1 = "testTitle1";
+  private static final String TITLE_2 = "testTitle2";
+  private static final String TITLE_3 = "testTitle3";
+
+  @Before
```

Comment:
You should have one test per public method. Your class has 5 public methods so I would expect 5 tests here. I would then expect a 6th test exercising the the IllegalArgument case

Reply:
+1